src/components/container/ProviderSearch/ProviderSearch.tsx

```diff
@@ -69,13 +71,31 @@ export default function ProviderSearch(props: ProviderSearchProps) {
         setSearching(true);
         let requestID = ++currentRequestID;
 
-        MyDataHelps.getExternalAccountProviders(search, props.providerCategories?.length == 1 ? props.providerCategories[0] : null, pageSize, currentPage).then(function (searchResultsResponse) {
-            if (requestID == currentRequestID) {
-                updateSearchResults(searchResultsResponse.externalAccountProviders);
-                setTotalResults(searchResultsResponse.totalExternalAccountProviders);
-                setSearching(false);
-            }
-        });
+        if (!props.publicProviderSearchApiUrl) {
+            MyDataHelps.getExternalAccountProviders(search, props.providerCategories?.length == 1 ? props.providerCategories[0] : null, pageSize, currentPage).then(function (searchResultsResponse) {
+                if (requestID == currentRequestID) {
+                    updateSearchResults(searchResultsResponse.externalAccountProviders);
+                    setTotalResults(searchResultsResponse.totalExternalAccountProviders);
+                    setSearching(false);
+                }
+            });
+        }
+        else {
+            fetch(`${props.publicProviderSearchApiUrl}?keyword=${search}&pageSize=${pageSize}&pageNumber=${currentPage}`, {
```

Comment:
Not only it is more readable, but also it handles parameter encoding for you.
```suggestion
      const url = new URL(props.publicProviderSearchApiUrl);
      url.searchParams.append('keyword', search);
      url.searchParams.append('pageSize', String(pageSize));
      url.searchParams.append('pageNumber', String(page));
      fetch(url.toString(), {
        method: "GET",
        headers: { "Accept": "application/json" },
      });
```

Reply:
Done